src/front-end/typescript/lib/pages/opportunity/helpers.tsx

```diff
@@ -0,0 +1,236 @@
+import React from 'react';
+import {CWUOpportunityStatus, CWUOpportunityEvent} from 'shared/lib/resources/opportunity/code-with-us';
+import {SWUOpportunityEvent, SWUOpportunityStatus} from 'shared/lib/resources/opportunity/sprint-with-us'
+import { toast} from 'front-end/lib/framework';
+import * as Attachments from 'front-end/lib/components/attachments'
+import * as api from 'front-end/lib/http/api';
+import { invalid } from 'shared/lib/validation';
+import { adt } from 'shared/lib/types';
+import { Alert } from 'reactstrap';
+
+
+export const MAX_NOTE_LENGTH = 1000;
+export const MAX_ATTACHMENT_SIZE = 2000000
+
+const toasts = {
+  success: {
+    title: 'Note Published',
+    body: 'Your note has been successfully published.'
+  },
+  error: {
+    title: 'Unable to Publish Note',
+    body: 'Your note could not be published. Please try again later.'
+  }
+};
+
+export function historyToHistoryTableRow(rawHistory){
+
+    const convertHistoryItemToHistoryTableItem = status =>{
```

Comment:
Colours and text from https://github.com/button-inc/digital_marketplace/blob/main/src/front-end/typescript/lib/pages/opportunity/code-with-us/lib/index.ts